library/core/src/time.rs

```diff
@@ -308,6 +308,29 @@ impl Duration {
         Duration { secs, nanos: subsec_nanos }
     }
 
+
+    /// creates a new Duration from the specified number of nano seconds. 
+    /// Use this function if you need to specify time greater than what can fit in u64 
+    /// which is around 584 years and above 584.94 years to be a little more precise. 
+    /// 
+    /// # Examples 
+    /// 
+    /// ```
+    /// use std::time::Duration;
+    /// let time_in_nanos = 2.pow(64);
+    /// let duration = Duration::from_nanos_u128(time_in_nanos);
+    /// 
+    /// ```
```

Comment:
You might want to add this stability attribute too.

```suggestion
    /// ```
    #[unstable(feature = "duration_from_nanos_u128", issue = "139201")]
    
```